### PR TITLE
Add Leaderboard feature

### PR DESCRIPTION
**Pull request by kaushikb11**
Built an isolated LeaderBoard Component which would take single or multiplayer usernames and scores by the props and update it to the leaderboard.

### CODE REVIEW

**Comment by kaushikb11** [comment]
Screenshot from 2019-03-16 03-14-15

**Comment by shivangg** [comment]
The commit message should be `feat: Add Leaderboard` rather than `feat: Leaderboard`. 

It should look like a command.

File: client/src/components/Layout/LeaderBoard.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+class LeaderBoard extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      leastHighScore: 0,
+      leaderBoardObject: {},
+      arrayOfLeaderBoardRows: [],
+    };
+  }
```
Comment (hassan23) [inline client/src/components/Layout/LeaderBoard.js:47]: Use useful names for variables:
`arrayOfLeaderBoardRows` can be declared as `LeaderBoardRowsList`

File: client/src/components/Layout/LeaderBoard.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  componentDidMount() {
+    this.newLeaderboardScore({
+      Hassan: 50, Shalini: 5, Shivang: 10, Mehul: 2, Kaushik: 101,
+    });
+  }
+
+  createData = (object) => {
+    let rank = 0;
+    const leaderBoardObjectRows = Object.entries(object)
+      .map(([username, score]) => {
```
Comment (nimish-gupta) [inline client/src/components/Layout/LeaderBoard.js:58]: are entries coming in sorted form?

File: client/src/components/Layout/LeaderBoard.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  row: {
+    '&:nth-of-type(odd)': {
+      backgroundColor: theme.palette.background.default,
+    },
+  },
+});
+
+
+class LeaderBoard extends Component {
+  constructor(props) {
```
Comment (nimish-gupta) [inline client/src/components/Layout/LeaderBoard.js:40]: class variables?

**Review by nimish-gupta — COMMENTED** [review]
LGTM

File: client/src/components/Layout/LeaderBoard.js (hunk 2 above)
Comment (kaushikb11) [reply client/src/components/Layout/LeaderBoard.js:58]: Yes.

File: client/src/components/Layout/LeaderBoard.js (hunk 3 above)
Comment (kaushikb11) [reply client/src/components/Layout/LeaderBoard.js:40]: The leaderboard component is isolated for now but it would take props.

**Review by hassan23 — APPROVED** [review]

**Review by nimish-gupta — APPROVED** [review]

**Pull request merged**